StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Utils/LinkURLGenerator.swift

```diff
@@ -4,36 +4,107 @@
 //
 
 import Foundation
+@_spi(STP) import StripeCore
+@_spi(STP) import StripePayments
 
 struct LinkURLParams: Encodable {
     struct MerchantInfo: Encodable {
-        let businessName: String
-        let country: String
+        var businessName: String
+        var country: String
     }
     struct CustomerInfo: Encodable {
-        let country: String
-        let email: String
+        var country: String
+        var email: String?
     }
     struct PaymentInfo: Encodable {
-        let currency: String
-        let amount: Int
-    }
-    let path = "mobile_pay"
-    let integrationType = "mobile"
-    let publishableKey: String
-    let merchantInfo: MerchantInfo
-    let customerInfo: CustomerInfo
-    let paymentInfo: PaymentInfo
-    let returnUrl: URL
-    let experiments: [String]
-    let flags: [String]
-    let loggerMetadata: [String]
-    let locale = Locale.current.toLanguageTag()
+        var currency: String
+        var amount: Int
+    }
+    enum PaymentObjectMode: String, Encodable {
+        case link_payment_method
+        case card_payment_method
+    }
+    var path = "mobile_pay"
+    var integrationType = "mobile"
+    var paymentObject: PaymentObjectMode
+    var publishableKey: String
+    var stripeAccount: String?
+    var paymentUserAgent: String
+    var merchantInfo: MerchantInfo
+    var customerInfo: CustomerInfo
+    var paymentInfo: PaymentInfo?
+    var experiments: [String: Bool]
+    var flags: [String: Bool]
+    var loggerMetadata: [String: Bool]
+    var locale: String
 }
 
 class LinkURLGenerator {
-    static func url() -> URL {
-//      TODO: Create URLs based on params
-        return URL(string: "https://checkout.link.com/#")!
+    static func linkParams(configuration: PaymentSheet.Configuration, intent: Intent) async throws -> LinkURLParams {
+        guard let publishableKey = configuration.apiClient.publishableKey ?? STPAPIClient.shared.publishableKey else {
+            throw LinkURLGeneratorError.noPublishableKey
+        }
+        guard let merchantCountryCode = intent.countryCode else {
+            throw LinkURLGeneratorError.noCountryCode
+        }
+
+        // US isn't the best default, but we only expect regionCode to be nil in the nil region on a simulator, and we don't want to crash.
```

Comment:
I think the wording is off:
"we only expect regionCode to be nil in the nil region on a simulator" ?

Also, is there some invalid country code that we could use instead of US?  I feel like if there are errors, then we'll never know about it?  Maybe "--" or worse case like "XX" (hopefully that won't make anything throw on the backend?)

Reply:
The simulator region is sometimes nil, which causes regionCode to be nil. That's the only situation in which I'd expect a nil region. (Unless someone blew up their `.GlobalPreferences.plist`.)
XX makes sense, though I do think it would cause a backend error. Still better than crashing.